Sources/BrowserServicesKit/SecureVault/SecureVaultCryptoProvider.swift

```diff
@@ -170,14 +181,14 @@ final class DefaultCryptoProvider: SecureVaultCryptoProvider {
         return nil
     }
 
-        
-    func hashData(_ data: Data) throws -> String? {
-        if let salt = self.salt {
-            let saltedData = salt + data
-            return SHA256.hash(data: saltedData).dataRepresentation.base64EncodedString()
+    func hashData(_ data: Data, salt: Data? = nil) throws -> String? {
+        guard let salt = salt ?? hashingSalt else {
+            throw SecureVaultError.failedToGetHashingSalt
```

Comment:
In previous implementation, salt could be nil and then this function would return nil, in the new implementation it throws an error. This will potentially have effect on the methods like `storeWebsiteCredentials` (breaking the flow instead of silently storing nil signature). I think it is a good change (as it falls under error handling we have there), just want to double check if you foresee any potential issues with that.

Reply:
@bwaresiak   I don't see any issues, but from a peace of mind (and since this is a hotfix), I can revert this change for release, and add it back after confirming everything is behaving correctly in the wild.  WDYT?